### PR TITLE
Julia changes related to the release of 1.7.0

### PR DESCRIPTION
**Pull request by ghost**
###### Motivation for this change

Julia 1.7.0 is the new stable and 1.6.4 the new LTS, with support for 1.0.5 being dropped. Reflect this in Nixpkgs.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/compilers/julia/1.7-bin.nix [hunk 1]
```diff
@@ -62,19 +61,11 @@ stdenv.mkDerivation rec {
   '';
 
   meta = {
-    description = "High-level, high-performance dynamic language for technical computing";
+    description = "High-level, high-performance dynamic language for technical computing.";
```
Comment (7c6f434c) [inline pkgs/development/compilers/julia/1.7-bin.nix:64]: https://nixos.org/manual/nixpkgs/stable/#var-meta-description

> Don’t include a period at the end.

**Comment by ninjin** [comment]
On Wed 01 Dec 2021, 7c6f434c wrote:
> 
> @7c6f434c commented on this pull request.
> 
> > @@ -62,19 +61,11 @@ stdenv.mkDerivation rec {
>    '';
>  
>    meta = {
> -    description = "High-level, high-performance dynamic language for technical computing";
> +    description = "High-level, high-performance dynamic language for technical computing.";
> 
> https://nixos.org/manual/nixpkgs/stable/#var-meta-description
> 
> > Don’t include a period at the end.

Well spotted, correcting for both `julia_16-bin` and `julia_17-bin`.

**Comment by 7c6f434c** [comment]
I wonder what are the chances that aarch64-linux will just work (say, at least build on ofborg)

**Comment by ninjin** [comment]
On Wed 01 Dec 2021, 7c6f434c wrote:
> 
> I wonder what are the chances that aarch64-linux will just work (say, at least build on ofborg)

Nearly 100% I would say, but I would either need to take the time to get my PINE64 up and running or have my PR merged for access to the community build box [1] to be able to debug. Still, I am a bit strapped for time at the moment, so I would greatly appreciate if this could be handled separately from this PR.

[1]: https://github.com/nix-community/aarch64-build-box/pull/109

**Comment by 7c6f434c** [comment]
Well, if there is need to debug beyond checking the build outcome on OfBorg, it would not be «just»

**Pull request merged**